profiles/views.py

```diff
@@ -127,3 +127,14 @@ class ViewedCompanyList(ListCreateAPIView):
     def get_queryset(self):
         user_id = self.request.user.id
         return ViewedCompany.objects.filter(user=user_id)
+
+
+class GetUserId(ListAPIView):
+    queryset = Profile.objects.all()
+    serializer_class = ProfileSerializer
+    permission_classes = (IsAuthenticated,)
+
+    def list(self, request):
+        profile = Profile.objects.get(person_id=self.request.user)
+        serializer = ProfileSerializer(profile)
+        return Response(serializer.data)
```

Comment:
``` python
class UserSelf(RetrieveAPIView):
    permission_classes = (IsAuthenticated,)
    serializer_class = ProfileSerializer

    def get_object(self):
        return self.request.user.profile
```

Reply:
Thank you for feedback. I have deleted it at all. I am using filtering for now